addons/link_tracker/models/link_tracker.py

```diff
@@ -68,12 +68,15 @@ def _compute_count(self):
     @api.depends('code')
     def _compute_short_url(self):
         for tracker in self:
-            base_url = self.env['ir.config_parameter'].sudo().get_param('web.base.url')
-            tracker.short_url = urls.url_join(base_url, '/r/%(code)s' % {'code': tracker.code})
+            tracker.short_url = urls.url_join(tracker.short_url_host, '%(code)s' % {'code': tracker.code})
 
     def _compute_short_url_host(self):
         for tracker in self:
-            tracker.short_url_host = self.env['ir.config_parameter'].sudo().get_param('web.base.url') + '/r/'
+            if self._context.get('website_id'):
```

Comment:
You can't mention anything related to website, or use website stuff (self.env['website'] etc) if you are not in a website dependant module.

This will need to go in an override of the compute in website_links